### PR TITLE
feat(UIForms): improve a11y

### PR DESCRIPTION
**Pull request by jsomsanith**
**What is the problem this PR is trying to solve?**
* errors are not assertive
* inputs are not linked to descriptions and errors
* no aria attributes

**What is the chosen solution to this problem?**
* make errors div always rendered, with `aria-live="assertive"`, so any change will be immediatly described by SR
* use aria-describedby to link input with the description and errors
* use aria-required on required values
* use aria-invalid on invalid inputs

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: output/forms.eslint.txt [hunk 1]
```diff
@@ -6,10 +6,12 @@ The react/require-extension rule is deprecated. Please use the import/extensions
...
 
 /home/travis/build/Talend/ui/packages/forms/src/UIForm/fields/CheckBox/SimpleCheckBox.component.js
-  9:4  error  Form controls using a label to identify them must be programmatically associated with the control using htmlFor  jsx-a11y/label-has-for
+  10:4  error  Form controls using a label to identify them must be programmatically associated with the control using htmlFor  jsx-a11y/label-has-for
+  11:5  error  The attribute aria-required is not supported by the role checkbox. This role is implicit on the element input    jsx-a11y/role-supports-aria-props
 
 /home/travis/build/Talend/ui/packages/forms/src/UIForm/fields/Radios/Radios.component.js
-  24:7  error  Form controls using a label to identify them must be programmatically associated with the control using htmlFor  jsx-a11y/label-has-for
+  25:7  error  Form controls using a label to identify them must be programmatically associated with the control using htmlFor  jsx-a11y/label-has-for
+  26:8  error  The attribute aria-required is not supported by the role radio. This role is implicit on the element input       jsx-a11y/role-supports-aria-props
```
Comment (talend-glorieux) [inline output/forms.eslint.txt]: Fix those errors.

File: packages/forms/src/UIForm/Message/generateId.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+	return id && `${id}-description`;
+}
+
+export function generateErrorId(id) {
+	return id && `${id}-error`;
+}
+
+export function generateDescribedBy(id) {
+	return id && `${generateDescriptionId(id)} ${generateErrorId(id)}`;
+}
```
Comment (talend-glorieux) [inline packages/forms/src/UIForm/Message/generateId.js]: Would be better if the method returned something meaningful if the id isn't passed.
In that case it returns `undefined` if no `id` is passed.

Comment (jsomsanith) [reply packages/forms/src/UIForm/Message/generateId.js]: I changed that to generate an id via uuid.v4() if there is no base id.
So each widget now generate it, and pass it to the input and FieldTemplate to set it on description and errors

File: packages/forms/src/UIForm/Message/generateId.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import uuid from 'uuid';
```
Comment (sgangnant) [inline packages/forms/src/UIForm/Message/generateId.js]: Since this is used by other form elements than Message, should't it be somewhere else ?

**Pull request merged**